### PR TITLE
Option for garbage collection logging & additional options

### PR DESCRIPTION
**Pull request by chrishenry**
There's no version bump here bc the maintainer prefers to do so himself.

### CODE REVIEW

**Comment by ko-be** [comment]
:+1:

File: templates/default/cassandra-env.sh.erb [hunk 1]
```diff
@@ -271,21 +271,23 @@ if [ "$JVM_VERSION_SHORT" = "1.7" ] && [ "$JVM_ARCH" = "64-Bit" ] ; then
...
+JVM_OPTS="$JVM_OPTS -XX:+PrintPromotionFailure"
+JVM_OPTS="$JVM_OPTS -XX:PrintFLSStatistics=1"
+JVM_OPTS="$JVM_OPTS -Xloggc:<%= node['cassandra']['log_dir'] -%>/gc-`date +%s`.log"
 # If you are using JDK 6u34 7u2 or later you can enable GC log rotation
 # don't stick the date in the log name if rotation is on.
-# JVM_OPTS="$JVM_OPTS -Xloggc:/var/log/cassandra/gc.log"
-# JVM_OPTS="$JVM_OPTS -XX:+UseGCLogFileRotation"
-# JVM_OPTS="$JVM_OPTS -XX:NumberOfGCLogFiles=10"
-# JVM_OPTS="$JVM_OPTS -XX:GCLogFileSize=10M"
+JVM_OPTS="$JVM_OPTS -Xloggc:/var/log/cassandra/gc.log"
```
Comment (markdunphy) [inline templates/default/cassandra-env.sh.erb]: You may want this to live in `/mnt/var/log/cassandra/gc.log` along with the `system.log`

File: templates/default/cassandra-env.sh.erb [hunk 2]
```diff
@@ -271,21 +271,23 @@ if [ "$JVM_VERSION_SHORT" = "1.7" ] && [ "$JVM_ARCH" = "64-Bit" ] ; then
...
-# JVM_OPTS="$JVM_OPTS -XX:PrintFLSStatistics=1"
-# JVM_OPTS="$JVM_OPTS -Xloggc:/var/log/cassandra/gc-`date +%s`.log"
+JVM_OPTS="$JVM_OPTS -XX:+PrintGCDetails"
+JVM_OPTS="$JVM_OPTS -XX:+PrintGCDateStamps"
+JVM_OPTS="$JVM_OPTS -XX:+PrintHeapAtGC"
+JVM_OPTS="$JVM_OPTS -XX:+PrintTenuringDistribution"
+JVM_OPTS="$JVM_OPTS -XX:+PrintGCApplicationStoppedTime"
+JVM_OPTS="$JVM_OPTS -XX:+PrintPromotionFailure"
+JVM_OPTS="$JVM_OPTS -XX:PrintFLSStatistics=1"
+JVM_OPTS="$JVM_OPTS -Xloggc:<%= node['cassandra']['log_dir'] -%>/gc-`date +%s`.log"
```
Comment (markdunphy) [inline templates/default/cassandra-env.sh.erb]: Comment this out again since you have `-XX:+UseGCLogFileRotation` enabled below

**Comment by mannytoledo** [comment]
:+1:

**Comment by mannytoledo** [comment]
@chrishenry any reason this can't be merged?

**Comment by chrishenry** [comment]
Nope, merge away. We're running it in prod right now. 

However, keep in mind that since this cookbook is OSS, I had to bump the version out of Source Control to upload, as we don't know when the maintainers will do a release.

For reference, this cookbook is on our [chef server as v0.5.3](https://chef.dev-be-aws.net/cookbooks/becassandra/0.5.3)

**Comment by ko-be** [comment]
@chrishenry I'd say we should make this PR metadata.rb the same version that's on our chef server.  But you pointed to becassandra, not cassandra.  Did you mean version `2.11.0`?

**Comment by chrishenry** [comment]
@ko-be oops, yea, I meant to link to the cassandra cookbook. I'd rather the leave the version the same as what's upstream, as we'll inevitably have to readjust once we contribute back / pull in upstream.

**Pull request merged**